modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -33,19 +31,17 @@ def do_remove(name, type)
       nil
     end
 
-    def execute(cmd, msg = nil, error_only = false)
+    def execute(*args, **keyword_args)
       tsecs = 5
       response = nil
-      interpreter = Proxy::SETTINGS.x86_64 ? 'c:\windows\sysnative\cmd.exe' : 'c:\windows\system32\cmd.exe'
-      command = interpreter + ' /c c:\Windows\System32\dnscmd.exe ' + "#{@server} #{cmd}"
-
       std_in = std_out = std_err = nil
+      real_cmd = ['c:\Windows\System32\dnscmd.exe' + "#{@server} #{args.join(' ')}"]
```

Comment:
Okay, now it's an array, but just with one element:

    ["command arg arg arg"]

It needs to be literally:

    ["command", "arg", "arg", "arg"]